File: java/parsers/src/main/java/jp/co/teruuu/parser/json/type/JString.java
package jp.co.teruuu.parser.json.type;

public class JString implements JValue{
    public String value;
    public JString(String value){
        this.value = value;
    }

    @Override
    public boolean equals(Object other) {
        return other instanceof JString && this.value.equals(((JString) other).value);
    }
}

File: scala/parsers/src/main/scala/jp/co/teruuu/parser/json/type/JString.scala
package jp.co.teruuu.parser.json.`type`

case class JString(value: String) extends JValue {

  override def equals(obj: Any): Boolean = {
    obj.isInstanceOf[JString] && value.equals(obj.asInstanceOf[JString].value)
  }

  def toSpray(): spray.json.JsValue = {
    spray.json.JsString(value)
  }
}
